src/Core/Utilities/JsonHelpers.cs

```diff
@@ -129,4 +130,63 @@ public override void Write(Utf8JsonWriter writer, DateTime? value, JsonSerialize
             writer.WriteStringValue(CoreHelpers.ToEpocMilliseconds(value.Value).ToString());
         }
     }
+
+    /// <summary>
+    /// Allows reading a string from a number or string, should only be used on string properties
+    /// </summary>
+    public class PermissiveStringConverter : JsonConverter<string>
+    {
+        internal static PermissiveStringConverter Instance = new();
+
+        public override string Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            return reader.TokenType switch
+            {
+                JsonTokenType.String => reader.GetString(),
+                JsonTokenType.Number => reader.GetInt64().ToString(),
+                JsonTokenType.Null => null,
+                _ => throw new JsonException($"Unsupported TokenType: {reader.TokenType}"),
+            };
+        }
+
+        public override void Write(Utf8JsonWriter writer, string value, JsonSerializerOptions options)
+        {
+            writer.WriteStringValue(value);
+        }
+    }
+
+    /// <summary>
+    /// Allows reading a string from a number or string, should only be used on string properties
```

Comment:
Please ensure if including XML comments for documentation that it's accurate and reflects the class or method appropriately. These don't mention that this should be used for `IEnumerable<string>` properties/implementations.